tests/Unit/Lock/lock_host_pointer.cpp

```diff
@@ -0,0 +1,24 @@
+// XFAIL: Linux
+// RUN: %hc %s -o %t.out %t.out
```

Comment:
@adityaatluri this line looks suspicious. The test is supposed to be linked with -lhc_am but it's not specified here.